### DISCUSSION TITLE
Make error when installing phoebe

### DISCUSSION THREAD

**Discussion by ypark90** [phoebe-team/phoebe#215] [Q&A] [answered]
I tried to install phoebe on ubuntu, but I couldn't.
Could I get some help resolving the problem? I attached the output files from cmake and make. 

[cmakeoutput.txt](https://github.com/user-attachments/files/16018876/cmakeoutput.txt)
[makeoutput.txt](https://github.com/user-attachments/files/16018877/makeoutput.txt)

Thank you.
YC

**Comment by jcoulter12** [comment]
Hi YC, 

Hm, looks like linking to libgfortran fails. Ubuntu 24.04 works in my git actions tests, so I think that there is something happening due to your specific Ubuntu environment. 

Probably we can figure it out. Can I ask why you use two different copies of GCC? 
```
-- The C compiler identification is GNU 12.3.0
-- The CXX compiler identification is GNU 11.4.0
-- The Fortran compiler identification is GNU 11.4.0
```

Best, 
Jenny

**Reply by ypark90** [reply]
Hi Jenny,
Thank you for your comment!

I initially didn’t realize I was using different versions of GCC, so I switched to using GCC version 11.4 uniformly, but the same error still occurred.
I will continue to find the linking issue with libgfortran.

Regards, 
YC

**Reply by jcoulter12** [reply]
Hi YC, 

No worries, feel free to ask more questions about this. 

Because of this line `/usr/bin/ld: /usr/local/lib/libscalapack.a(pxerbla.o): undefined reference to symbol '_gfortran_transfer_integer_write@@GFORTRAN_8'`

I would wonder if basically the copy of libscalapack found in your usr/local/lib folder is built against a different version of GCC than the one being found by CMake for the build, or something like this. I would try to figure out where that lib file came from. 

Another option is to allow Phoebe to build its own copy of ScaLAPACK. If you can remove that ScaLAPACK, so that Phoebe doesn't find any version of ScaLAPACK in the CMake process, it will pull down and build a copy from github. 

For an additional hint, in our build and test, we use this line to set up the proper dependencies for Ubuntu when using 22.04:

```
sudo apt-get update
sudo apt install -y libopenblas-dev liblapack-dev libhdf5-dev libhdf5-openmpi-dev libscalapack-openmpi-dev python3-pip gfortran cmake -o Acquire::Retries=3 --fix-missing
```

Let me know if I can answer any further questions on this, 
Jenny

**Reply by ypark90** [reply, marked as answer]
Hi Jenny,
Thank you for your valuable comments!

I rebuilt the copy of libscalapack with the proper GCC version, but the same error still occurred. 
Fortunately, I resolved the linking issue by using the following option, although I’m not sure if it is the proper solution:

`export LDFLAGS="-Wl,--copy-dt-needed-entries"
`

With this change, I successfully generated the executable Phoebe in the build directory.

Thank you!
YC

**Reply by jcoulter12** [reply]
Hi YC, 

Thanks for reporting back with the solution, and feel free to open another discussion if you have questions about using the code. 

Best, 
Jenny
